kubectl-openebs/cli/command/get/volume_list.go

```diff
@@ -79,16 +79,30 @@ func RunVolumesList(cmd *cobra.Command, openebsNs string, vols []string) error {
 	// give output according to volume status
 	var rows []metav1.TableRow
 	for _, item := range cvols.Items {
-		rows = append(rows, metav1.TableRow{Cells: []interface{}{
-			item.ObjectMeta.Namespace,
-			item.ObjectMeta.Name,
-			item.Status.Phase,
-			item.VersionDetails.Status.Current,
-			util.ConvertToIBytes(item.Status.Capacity.String()),
-			pvols[item.ObjectMeta.Name].StorageClass,
-			pvols[item.ObjectMeta.Name].AttachementStatus,
-			pvols[item.ObjectMeta.Name].AccessMode,
-			pvols[item.ObjectMeta.Name].Node}})
+		if val, ok := pvols[item.ObjectMeta.Name]; ok {
+			rows = append(rows, metav1.TableRow{Cells: []interface{}{
+				item.ObjectMeta.Namespace,
+				item.ObjectMeta.Name,
+				item.Status.Phase,
+				item.VersionDetails.Status.Current,
+				util.ConvertToIBytes(item.Status.Capacity.String()),
+				val.StorageClass,
+				val.AttachementStatus,
+				val.AccessMode,
+				val.Node}})
+		} else {
+			rows = append(rows, metav1.TableRow{Cells: []interface{}{
+				item.ObjectMeta.Namespace,
+				item.ObjectMeta.Name,
+				item.Status.Phase,
+				item.VersionDetails.Status.Current,
+				util.ConvertToIBytes(item.Status.Capacity.String()),
+				"N/A",
+				"N/A",
```

Comment:
(low-priority) This can be set to `NotAttached`, need to check if Kubernetes has a term or a state for this.

Reply:
- Yep, adding to constants.